src/kbmod/search/stack_search.cpp

```diff
@@ -155,6 +155,26 @@ Trajectory StackSearch::search_linear_trajectory(short x, short y, float vx, flo
     return result;
 }
 
+void StackSearch::finish_search(){
+    psi_phi_array.clear_from_gpu();
+    gpu_search_list.move_to_cpu();
+}
+
+void StackSearch::prepare_batch_search(std::vector<Trajectory>& search_list, int min_observations){
+    DebugTimer psi_phi_timer = DebugTimer("Creating psi/phi buffers", rs_logger);
+    prepare_psi_phi();
+    psi_phi_array.move_to_gpu();
+    psi_phi_timer.stop();
+
+    
+    int num_to_search = search_list.size();
+    if (debug_info) std::cout << "Preparing to search " << num_to_search << " trajectories... \n" << std::flush;
+    gpu_search_list = TrajectoryList(search_list);
```

Comment:
This is the only reason why I implemented the move assignment operators. Note sure if there's a neater/better way to not have to do it.

Reply:
Couldn't we just do `gpu_search_list.set_trajectories(search_list)` here? `set_trajectories()` should take care of resizing the memory allocations and copying over the new data.

Reply:
Yeah - that's cleaner and less error prone. Adapted.